typings/mapping.d.ts

```diff
@@ -0,0 +1,8 @@
+/** Generated from {@link ../tools/create-mapping.sh} */
+import { LitElement } from 'lit';
+declare global {
+	interface HTMLElementTagNameMap {
+		// Components with export class...extends LitElement
+		'd2l-build-info': import('../index.js').BuildInfo;
```

Comment:
Just starting with `d2l-build-info` as a proof of concept. Once the other PRs: https://github.com/BrightspaceUI/core/pull/5140, https://github.com/BrightspaceUI/core/pull/5141 are merged, the mapping.d.ts can be rebuilt.

Reply:
After this, a CI step can be created to automatically update the mapping as needed. Then, adding new components to the map should be very simple. Just add an export or `@typedef`, then a PR will automatically be created to add it, and the lint step will ensure that the mixins are being used correctly.